jax/_src/numpy/lax_numpy.py

```diff
@@ -3555,8 +3571,25 @@ def fromstring(string: str, dtype: DTypeLike = float, count: int = -1, *, sep: s
   return asarray(np.fromstring(string=string, dtype=dtype, count=count, sep=sep))
```

Comment:
Seems like maybe we should add `device` support to `asarray` first, and then this change would be much simpler.